web_api/national_park/national_parks.py

```
@@ -0,0 +1,57 @@
+import requests
+import json
+from park import Park
+
+# National Park Service API endpoint
+url = 'https://developer.nps.gov/api/v1/parks'
+
+# API key ( I stored on my pc )
+api_key = 'NPS_API_KEY'
+
+"""
+Defined two separate functions: make_request() and process_data(). The make_request() function takes a state_code parameter, sends an API request to the National Park Service API,
+ and returns the response as JSON. This function includes error handling for cases where the request fails or the response is not in JSON format.
+"""
+def make_requests(state_code):
+    """ Makes an API request to the National Park Service API and returns the response as JSON."""
+    params = {'api_key': api_key, 'stateCode': state_code}
+    try:
+        response = requests.get(url, params=params)
+        response.raise_for_status() # Raise an exception if the response status code is not 200 OK
+        return response.json()
+    except (requests.exceptions.RequestException, json.JSONDecodeError) as e:
+        print('Error: ', e)
```

Maybe you already knew this and it was just for testing, but in the production version of this code this should return the error code/message or log it rather than print.